src/Config.php

```diff
@@ -93,11 +95,15 @@ private function setConsumerGroup(array $topicConfig, string $consumerGroupId):
 
     private function setBroker(array $topicConfig): void
     {
-        $this->broker = config("kafka.brokers.{$topicConfig['broker']}");
+        $brokerConfig = config("kafka.brokers.{$topicConfig['broker']}");
 
-        if (!$this->broker) {
+        if (!$brokerConfig) {
             throw new ConfigurationException("Broker '{$topicConfig['broker']}' configuration not found");
         }
+
+        $this->broker = new Broker();
+        $this->broker->setConnection($brokerConfig['connection']);
+        $this->broker->setAuthentication($brokerConfig['auth']);
```

Comment:
I'll explore this in the tests scenarios, but because the `setAuthentication` argument is optional (falling to null when no 'auth' key exists), I think that it would be the case that is a connection with no authentication needed